### ISSUE TITLE
Continue on fail not working with parameter agregation

### ISSUE THREAD

**Issue by nenadzaric** [argoproj/argo-workflows#2743]
Checklist:

 * [x] I've included the version.
 * [x] I've included reproduction steps.
 * [x] I've included the workflow YAML.
 * [ ] I've included the logs.
 
**What happened**:

Dag stops if one of the steps made with parameter aggregation is failed. For example, if you pass 2 parameters to the template, for two different executions, and one fails, DAG will not execute next step (whalesay in example) even with continueOn directive.

**What you expected to happen**:

Whalesay to be executed after one-will-fail.

**How to reproduce it (as minimally and precisely as possible)**:
```
apiVersion: argoproj.io/v1alpha1
kind: Workflow
metadata:
  generateName: parameter-aggregation-one-will-fail-
spec:
  entrypoint: parameter-aggregation-one-will-fail
  templates:
  - name: parameter-aggregation-one-will-fail
    dag:
      tasks:
      - name: one-will-fail
        template: one-will-fail
        arguments:
          parameters:
          - name: num
            value: "{{item}}"
        withItems: [1, 2]
        continueOn:
          failed: true

      - name: whalesay
        template: whalesay
        dependencies:
          - one-will-fail
        continueOn:
          failed: true

  - name: one-will-fail
    inputs:
      parameters:
      - name: num
    container:
      image: alpine:latest
      command: [sh, -xc]
      args:
      - |
        if [ $(({{inputs.parameters.num}})) == 1 ]; then
          exit 1;
        else
          echo {{inputs.parameters.num}}
        fi

  - name: whalesay
    container:
      image: docker/whalesay:latest
      command: [cowsay]
```
**Anything else we need to know?**:

**Environment**:
- Argo version:
```
$ argo version
argo: v2.7.3
  BuildDate: 2020-04-15T22:52:59Z
  GitCommit: 66bd0425280c801c06f21cf9a4bed46ee6f1e660
  GitTreeState: clean
  GitTag: v2.7.3
  GoVersion: go1.13.4
  Compiler: gc
  Platform: darwin/amd64
```
- Kubernetes version :
```
$ kubectl version -o yaml
clientVersion:
  buildDate: "2019-10-15T19:16:51Z"
  compiler: gc
  gitCommit: 20c265fef0741dd71a66480e35bd69f18351daea
  gitTreeState: clean
  gitVersion: v1.15.5
  goVersion: go1.12.10
  major: "1"
  minor: "15"
  platform: darwin/amd64
serverVersion:
  buildDate: "2020-03-12T20:55:23Z"
  compiler: gc
  gitCommit: 8d8aa39598534325ad77120c120a22b3a990b5ea
  gitTreeState: clean
  gitVersion: v1.17.4
  goVersion: go1.13.8
  major: "1"
  minor: "17"
  platform: linux/amd64
```

**Other debugging information (if applicable)**:
- workflow result:
```
argo get parameter-aggregation-one-will-fail-rnxpj
Name:                parameter-aggregation-one-will-fail-rnxpj
Namespace:           default
ServiceAccount:      default
Status:              Error
Message:             task from node parameter-aggregation-one-will-fail-rnxpj.one-will-fail(0:1) does not exist
Created:             Sat Apr 18 00:32:14 +0200 (5 seconds ago)
Started:             Sat Apr 18 00:32:14 +0200 (5 seconds ago)
Finished:            Sat Apr 18 00:32:18 +0200 (1 second ago)
Duration:            4 seconds

STEP                                          TEMPLATE                             PODNAME                                               DURATION  MESSAGE
 ⚠ parameter-aggregation-one-will-fail-rnxpj  parameter-aggregation-one-will-fail                                                                  Workflow operation error
 ├-✖ one-will-fail(0:1)                       one-will-fail                        parameter-aggregation-one-will-fail-rnxpj-1687830053  2s        failed with exit code 1
 └-✔ one-will-fail(1:2)                       one-will-fail                        parameter-aggregation-one-will-fail-rnxpj-2531028903  2s
```
---

**Message from the maintainers**:

If you are impacted by this bug please add a 👍 reaction to this issue! We often sort issues this way to know what to prioritize.

**Comment by simster7** [comment]
Thanks for this. Opened #2756 to fix this.

**Comment by simster7** [comment]
The fix for this has been released in v2.7.5: https://github.com/argoproj/argo/releases/tag/v2.7.5
